cmd/hdfs/put.go

```diff
@@ -29,6 +31,48 @@ func put(args []string) {
 		fatal(err)
 	}
 
+	if filepath.Base(source) == "-" {
+		putFromStdIn(client, dest)
+	} else {
+		putFromFile(client, source, dest)
+	}
+}
+
+func putFromStdIn(client *hdfs.Client, dest string) {
```

Comment:
Nitpick: this should be `putFromStdin` (without the `I` capitalized), since the standard library spells it that way.